impeller/renderer/backend/vulkan/render_pass_vk.cc

```diff
@@ -399,27 +404,27 @@ bool RenderPassVK::SetIndexBuffer(BufferView index_buffer,
   if (index_type != IndexType::kNone) {
     has_index_buffer_ = true;
 
-    const BufferView& index_buffer_view = index_buffer;
+    BufferView index_buffer_view = std::move(index_buffer);
     if (!index_buffer_view) {
       return false;
     }
 
-    const std::shared_ptr<const DeviceBuffer>& index_buffer =
-        index_buffer_view.buffer;
-    if (!index_buffer) {
+    if (!index_buffer_view.GetBuffer()) {
       VALIDATION_LOG << "Failed to acquire device buffer"
                      << " for index buffer view";
       return false;
     }
 
-    if (!command_buffer_->Track(index_buffer)) {
+    std::shared_ptr<const DeviceBuffer> index_buffer =
```

Comment:
So we only track it if the buffer_view owns the device buffer reference? Oh and then we set the buffer view to use a raw_ptr as we know its tracked already...

Incredibly clever!